StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs

```diff
@@ -66,14 +66,32 @@ internal static StyleCopSettings GetStyleCopSettings(this AnalyzerOptions option
             return GetStyleCopSettings(options != null ? options.AdditionalFiles : ImmutableArray.Create<AdditionalText>(), failureBehavior, cancellationToken);
         }
 
+        /// <summary>
+        /// Gets a value indicating whether the specified path points to a StyleCop settings file (stylecop.json or .stylecop.json).
+        /// </summary>
+        /// <param name="path">The path to test.</param>
+        /// <returns><c>true</c> if <paramref name="path"/> points to a StyleCop settings file; otherwise, <c>false</c>.</returns>
+        internal static bool IsStyleCopSettingsFile(string path)
+        {
+            var fileName = Path.GetFileName(path).ToLowerInvariant();
```

Comment:
This also allows `stylecop.json` and `.stylecop.json` files in folders other than the root folder of the project, which is a change from the current behavior.

Reply:
It would also be better to use a case insensitive string compare, instead of converting to lower case first, as this saves additional memory allocations.

Reply:
> This also allows stylecop.json and .stylecop.json files in folders other than the root folder of the project, which is a change from the current behavior.

This is already indirectly possible using linked files.

Reply:
:memo: This approach appears to be a simple preservation of the original behavior.

Reply:
The string comparison is changed. This change did cause a unit test to fail which expected a NullReferenceException to be unhandled (when path is null, calling .ToLowerInvariant() causes an NullReferenceException). I've added a null check and now an ArgumentNullException is thrown when the path is null (the test works again).